enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,364 @@
+- **Resource Usage**: The proxy's resource requirements will be minimal, as it only handles HTTP01 challenge traffic.
+- **Failure Recovery**: Health checks will ensure that the proxy is running correctly, and failed pods will be automatically restarted.
+
+#### Recovery Procedures
+
+If the proxy DaemonSet enters a `CrashLoopBackOff` state, HTTP01 challenges for the API endpoint will fail, and certificate renewal will not complete. This may result in the API certificate expiring, which could impact cluster operations.
+
+**Recovery steps:**
+- Cluster administrators can disable or remove the proxy by deleting the DaemonSet or updating the relevant CR/manifest.
+- If the API certificate has expired, recovery may require connecting to the cluster's API server while ignoring certificate validation errors (e.g., using `--insecure-skip-tls-verify` with `oc` or `kubectl`).
+- After resolving the issue (e.g., fixing the DaemonSet, node configuration, or proxy image), re-deploy the proxy to restore HTTP01 challenge functionality.
+- The proxy should surface clear status and error messages to help identify and resolve CrashLoopBackOff or degraded states.
+
+## Support Procedures
+
+### Detecting Failure Modes
+
+- **Symptoms**: 
+  - Cert Manager HTTP01 challenges for the API endpoint (`api.cluster.example.com`) fail to complete.
+  - Certificates for the API endpoint are not issued or renewed.
+  - The proxy DaemonSet pods are in `CrashLoopBackOff` or `Error` state.
+  - Events in the `openshift-cert-manager` or relevant namespace indicate pod failures.
+  - The operator managing the proxy (if any) reports a degraded or error status.
+  - Logs from the proxy pod show errors related to `nftables` or port binding.
+  - The API server logs may show failed ACME challenge requests or timeouts.
+
+- **Metrics/Alerts**:
+  - Custom metrics (if implemented) such as `cert_manager_proxy_up` or `cert_manager_proxy_errors_total` may indicate proxy health.
+  - Alerts can be configured for DaemonSet unavailability or excessive restarts.
+
+### Disabling the API Extension
+
+- **How to disable**: 
+  - Remove or scale down the proxy DaemonSet.
+  - Remove or update the associated CR (if using a CRD).
+  - Remove any MachineConfig or configuration that enables the proxy.
+- **Consequences**:
+  - HTTP01 challenges for the API endpoint will not be possible.
+  - Certificates for the API endpoint will not be issued or renewed.
+  - If the API certificate expires, cluster API access may be impacted until a valid certificate is restored.
+
+### Impact on Existing, Running Workloads
+
+- Existing workloads and API traffic will continue to function as long as the API certificate is valid.
+- If the API certificate expires and is not renewed, clients may fail to connect to the API server due to certificate errors.
+- No direct impact on running pods or services, unless they rely on the API endpoint with a valid certificate.
```

Comment:
API endpoint meaning the public route to the API endpoint? Just wanting to verify that this shouldn't have any impact to internal DNS that operator pods, for example, may use.